core-bundle/src/EventListener/DataContainer/TextTrackFieldsListener.php

```diff
@@ -0,0 +1,34 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\EventListener\DataContainer;
+
+use Contao\CoreBundle\DataContainer\PaletteManipulator;
+use Contao\CoreBundle\DependencyInjection\Attribute\AsCallback;
+use Contao\DataContainer;
+
+class TextTrackFieldsListener
+{
+    #[AsCallback(table: 'tl_files', target: 'config.onpalette')]
+    public function addTextTrackFields(string $palette, DataContainer $dc): string
+    {
+        // $dc->id is the file name in this case
+        if (str_ends_with($dc->id, '.vtt')) {
```

Comment:
Nitpick: This comparison should probably use `Path::getExtension()` with `forceLowercase` set to `true`. Otherwise for example a `foo.VTT` file would not be recognized.

Reply:
See 10fe148937804ff1b5068b57d2ffd5777eac707b

Reply:
then obviously without the `str_ends_with` :smile:

Reply:
OOOH :D